app/code/Magento/Catalog/Pricing/Render/FinalPriceBox.php

```diff
@@ -86,6 +88,15 @@ protected function _toHtml()
         return $this->wrapResult($result);
     }
 
+    /**
+     * @param SaleableInterface $salableItem
+     * @return bool
+     */
+    protected function isApplySalableCheck(SaleableInterface $salableItem): bool
+    {
+        return (bool)$salableItem;
```

Comment:
Object will be always converted to `true`, isn't it?

Reply:
Yes, I’ve missed it. This method from Catalog module should always return true.
Will change it tomorrow.

Reply:
I've removed unneeded usage of the variable. However, there was a need to add `@SuppressWarnings(PHPMD.UnusedFormalParameter)` phpdoc for `isApplySalableCheck` method to make it pass static tests.

Goal of this method `isApplySalableCheck`: make price rendering depending not only on `isSalable` check but also on addaitional check that can be modified in child classes without need to rewrite [_toHtml ](https://github.com/vpodorozh/magento2/blob/207e2e14306ff1ebea3294d82ed47b8804778d73/app/code/Magento/Catalog/Pricing/Render/FinalPriceBox.php#L66)method.
I've used this "newly added" posibility in this class [Magento\ConfigurableProduct\Pricing\Render\FinalPriceBox ](https://github.com/vpodorozh/magento2/blob/207e2e14306ff1ebea3294d82ed47b8804778d73/app/code/Magento/ConfigurableProduct/Pricing/Render/FinalPriceBox.php#L95)

An other way to achieve this goal is to rewrite _toHtml method that will require a big amount of copy&paste code due to `private` properties and methods usage.

So I do prefer currently implemented solution.